packages/office-ui-fabric-react/src/components/Fabric/Fabric.base.tsx

```diff
@@ -13,37 +12,58 @@ export class FabricBase extends React.Component<
 > {
   private _rootElement = React.createRef<HTMLDivElement>();
   private _disposables: (() => void)[] = [];
+  private _removeClassNameFromBody?: () => void = undefined;
 
   constructor(props: IFabricProps) {
     super(props);
     this.state = { isFocusVisible: false };
   }
 
   public render() {
-    const { className } = this.props;
-
-    const classNames = getClassNames(getStyles, {
-      theme: this.props.theme!,
-      className,
-      isFocusVisible: this.state.isFocusVisible
-    });
+    const classNames = this._getClassNames();
     const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties);
-
     return <div {...divProps} className={classNames.root} ref={this._rootElement} />;
   }
 
   public componentDidMount(): void {
     const win = getWindow(this._rootElement.current);
-
     if (win) {
       this._disposables.push(on(win, 'mousedown', this._onMouseDown, true), on(win, 'keydown', this._onKeyDown, true));
     }
+    this._addClassNameToBody();
   }
 
   public componentWillUnmount(): void {
     this._disposables.forEach((dispose: () => void) => dispose());
+    if (this._removeClassNameFromBody) {
+      this._removeClassNameFromBody();
+    }
   }
 
+  private _getClassNames(): IProcessedStyleSet<IFabricStyles> {
+    const { className, theme, applyTheme } = this.props;
+    const classNames = getClassNames(getStyles, {
+      theme: theme!,
+      applyTheme: applyTheme,
+      className,
+      isFocusVisible: this.state.isFocusVisible
+    });
+    return classNames;
+  }
+
+  private _addClassNameToBody = () => {
```

Comment:
any reason why this is a lambda?

Reply:
Since it's using `this` inside a couple times, to give it the right context of `this` and since no arguments are being used.

Reply:
Using a lambda to ensure the correct `this` is only necessary if the method will be passed as a callback. If it's only ever called directly (`this._addClassNameToBody()`), it's not necessary to use a lambda, since `this` inside `_addClassNameToBody` will automatically be set to the object the method was called on. (The "as an object method" section of [this article](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/this) explains it better.) And the number of arguments doesn't really make a difference for whether something should be a lambda or not.

Overall, while it's not technically wrong to use a lambda here (everything will still work as expected as far as I'm aware), it does have the slight downside of implying that the method is passed as a callback, when in fact that's not the case.

Reply:
Thanks for this clarification elizabeth! I got rid of the lambda.